src/guest_memory.rs

```diff
@@ -73,6 +73,33 @@ pub enum Error {
     HostAddressNotAvailable,
 }
 
+impl PartialEq for Error {
+    fn eq(&self, other: &Self) -> bool {
+        match (self, other) {
+            (Error::InvalidGuestAddress(left), Error::InvalidGuestAddress(right)) => left == right,
+            (Error::InvalidBackendAddress, Error::InvalidBackendAddress) => true,
+            (Error::HostAddressNotAvailable, Error::HostAddressNotAvailable) => true,
+            (
+                Error::PartialBuffer {
+                    expected: left_expected,
+                    completed: left_completed,
+                },
+                Error::PartialBuffer {
+                    expected: right_expected,
+                    completed: right_completed,
+                },
+            ) => left_expected == right_expected && left_completed == right_completed,
+            (Error::IOError(left), Error::IOError(right)) => {
+                // error.kind should be enough to assert equallity because each error
```

Comment:
While this is true, it does not mean that `kind` can be used to compare errors. The reason why io::Error does not have PartialEq implemented is because it cannot be implemented truthfully. The first example that comes to mind is `NotFound`. A not found error thrown for `file1` is not equal to an error thrown for `file2`. This is one of the reasons why we cannot actually compare the `io::Error` like that because it has other metadata that is specific to the error type. We can say instead that we are willing to accept the loss of information in comparing errors because vm-memory errors are widely used in other crates as well and it is very limiting to not have `PartialEq` implemented.